fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -624,6 +691,23 @@ private boolean deleteFileInternal(@Nonnull Map<String, FDBLuceneFileReference>
         context.ensureActive().clear(metaSubspace.pack(name));
         // Nothing stored here currently.
         context.ensureActive().clear(dataSubspace.subspace(Tuple.from(value.getId())).range());
+        // Delete K/V data: If the deferredDelete flag is on then delete all content from K/V subspace for the segment
+        // (this is to support CFS deletion, that will be disjoint from the actual file deletion).
+        // Otherwise, delete the data for the specific file immediately.
+        String segmentName = IndexFileNames.parseSegmentName(name);
+        if (deferDeleteToCompoundFile) {
```

Comment:
I don't quite follow that one - if the CFS was deleted because this is an abort, then should there be any data saved at all?

Reply:
I'd have to go back to that and investigate what was happening. I _suspect_ that it will only delete the cfs in situations where it will error, and thus not commit, but I'm not sure. I don't think we need to block this change here, but something we'll need to validate.